### PR TITLE
Ignore unknown field when parsing proxy config.

### PR DESCRIPTION
**Pull request by yusuoh**
**What this PR does / why we need it**:
Ignore unknown protobuf field

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes https://github.com/istio/istio/issues/3428

**Release note**:

```release-note
```

### CODE REVIEW

File: src/envoy/mixer/utils_test.cc [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+#include "src/envoy/mixer/utils.h"
+#include "mixer/v1/config/client/client_config.pb.h"
+#include "test/test_common/utility.h"
+
+using Envoy::Http::Utils::ParseJsonMessage;
+
+namespace {
+
+TEST(UtilsTest, ParseNormalMessage) {
+  std::string config_str =
```
Comment (lizan) [inline src/envoy/mixer/utils_test.cc]: Use raw string literal (i.e. `R"({...})"` so it is easier to read. JSON keys need `"`s as well.

File: src/envoy/mixer/config.cc [hunk 2]
```diff
@@ -93,8 +94,7 @@ bool ReadV2Config(const Json::Object& json, Message* message) {
     return false;
   }
   std::string v2_str = json.getObject(kV2Config)->asJsonString();
-  Status status =
-      ::google::protobuf::util::JsonStringToMessage(v2_str, message);
+  Status status = Utils::ParseJsonMessage(v2_str, message);
```
Comment (lizan) [inline src/envoy/mixer/config.cc]: I think we should just use `MessageUtil::loadFromJson` from envoy, the behavior should be consistent with it.

Comment (yusuoh) [reply src/envoy/mixer/config.cc]: That one is relying on a third-party lib to parse the json. I would rather ensure that the option is set correctly by using the protobuf lib directly, as the logic is simple here and shouldn't involve much overhead.

File: src/envoy/mixer/utils_test.cc (hunk 1 above)
Comment (yusuoh) [reply src/envoy/mixer/utils_test.cc]: Done

File: src/envoy/mixer/config.cc (hunk 2 above)
Comment (lizan) [reply src/envoy/mixer/config.cc]: No it doesn't use third-party lib to parse:
https://github.com/envoyproxy/envoy/blob/910058d7113e6439d077e2467e26abb7b803000e/source/common/protobuf/utility.cc#L38

Do we see any real problem with unknown fields? If you don't fix upstream, the issue could be still there.

Comment (yusuoh) [reply src/envoy/mixer/config.cc]: Then it could also be an issue for envoy. I will check the upstream.

**Comment by yusuoh** [comment]
@lizan Could you please LGTM this PR for now to unblock the failed upgrade test? I will investigate the upstream issue separately.

**Comment by yusuoh** [comment]
A little more context, this issue happens during upgrade phase when the envoy proxy receives a JSON config from Pilot which has a different version with the proxy.

**Pull request merged**